src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -186,280 +214,319 @@ def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
-                            # it.
-                            if set_idx is None:
-                                unique_id = self.next_slice_id()
-                                unique_alternate_id = unique_id
-                                open_ = m_open.group(1)
-                                close_ = m_close.group(1)
-                                alternate_code = (
-                                    f"\0{unique_alternate_id} {open_} "
-                                    f"{''.join(trimmed_parts)} {close_}"
-                                )
-                if block_type == "block_start" and trimmed_parts[0] in (
-                    "macro",
-                    "set",
-                ):
-                    # Jinja supports two forms of {% set %}:
-                    # - {% set variable = value %}
-                    # - {% set variable %}value{% endset %}
-                    # https://jinja.palletsprojects.com/en/2.10.x/templates/#block-assignments
-                    # When the second format is used, set the variable 'set_idx'
-                    # to a non-None value. This info is used elsewhere, as
-                    # literals inside a {% set %} block require special handling
-                    # during the trace.
-                    filtered_trimmed_parts = [
-                        p for p in trimmed_parts if not p.isspace()
-                    ]
-                    if (
-                        len(filtered_trimmed_parts) < 3
-                        or filtered_trimmed_parts[2] != "="
-                    ):
-                        set_idx = len(result)
-                elif (
-                    block_type == "block_end"
-                    and set_idx is not None
-                    and (trimmed_parts[0] in ("endmacro", "endset"))
-                ):
-                    # Exiting a {% set %} block. Clear the indicator variable.
-                    set_idx = None
-                m = regex.search(r"\s+$", raw, regex.MULTILINE | regex.DOTALL)
-                if raw.startswith("-") and m:
+                        tag_contents = self.extract_tag_contents(
+                            str_parts, m_close, m_open, str_buff
+                        )
+
+                    if block_type == "block" and tag_contents:
+                        block_type, block_subtype = self.extract_block_type(
+                            tag_contents[0], block_subtype
+                        )
+                    if block_type == "templated" and tag_contents:
+                        assert m_open and m_close
+                        raw_slice_info = self.track_templated(
+                            m_open, m_close, tag_contents
+                        )
+                self.update_inside_set_or_macro(block_type, tag_contents)
+                m_strip_right = regex.search(
+                    r"\s+$", raw, regex.MULTILINE | regex.DOTALL
+                )
```

Comment:
What happening here ?

Reply:
Not sure you need the `DOTALL` here (there is no `.`)?

Reply:
We're trying to determine if a block uses right whitespace stripping, e.g.
```
SELECT
  {{ 'col1,' -}}
  col2
```

Note the `-`. This tells Jinja to discard any whitespace between the `}}` and the next bit of text. We need to account for this because it affects the template output.

Reply:
I can't remember why these flags were added. Maybe they were necessary back when we only had the whole block string, rather than the parts? Note that in dbt, it's common to do things like:
```
```

`DOTALL` actually means "allow the `.` pattern character to match newlines`, and `MULTILINE` means "allow multiline matches". https://www.codespeedy.com/re-dotall-in-python/

I'm hesitant to change this now. 😅  If the string were really long, I could see it slowing things down a bit, but this string is literally just one Jinja token -- should be very short.

Reply:
Fair point. It' snot changed in this PR and don't think it'll matter too much one way of another. So fine leaving it.